examples/siro_sandbox/controllers.py

```diff
@@ -423,25 +422,6 @@ def _update_grasp(self, grasp_object_id, drop_pos):
             )
             rigid_obj.translation = drop_pos
 
-    def _update_controller_to_navmesh(self):
-        # prevents humanoid agent from stepping through the walls and obstacles
-        # https://github.com/facebookresearch/habitat-lab/blob/2b7eaf9266a8916c0851c79043236ff2e55eaca0/habitat-lab/habitat/tasks/rearrange/actions/oracle_nav_action.py#L153
-        articulated_agent = self.get_articulated_agent()
-        trans = articulated_agent.sim_obj.transformation
-        rigid_state = habitat_sim.RigidState(
-            mn.Quaternion.from_matrix(trans.rotation()), trans.translation
-        )
-        target_rigid_state_trans = (
-            self._humanoid_controller.obj_transform_base.translation
-        )
-        end_pos = self._env._sim.step_filter(
```

Comment:
Here's roughly the problem with how we were calling this:
* Start and end positions are at pelvis height, not feet/navmesh height. This is generally risky.
* Normally, `step_filter`'s return value is a position that has been snapped down to the navmesh.
    * Hence the need to do `end_pos -= articulated_agent.params.base_offset` to get a translation suitable to assign to `self._humanoid_controller.obj_transform_base.translation` (i.e. raise `end_pos` back to pelvis height).
* However, sometimes start or end positions can't be mapped to the navmesh.
    * This happens often in the sandbox app, namely when you try to walk into a wall.
    * In this case, `step_filter` just returns the start pos, which was at pelvis height. So, it is incorrect to apply `articulated_agent.params.base_offset` in that case (`end_pos` gets raised above pelvis height).